staging/src/k8s.io/component-helpers/scheduling/corev1/helpers.go

```diff
@@ -43,3 +43,50 @@ func MatchNodeSelectorTerms(
 	}
 	return nodeaffinity.NewLazyErrorNodeSelector(nodeSelector).Match(node)
 }
+
+// TolerationsTolerateTaint checks if taint is tolerated by any of the tolerations.
+func TolerationsTolerateTaint(tolerations []v1.Toleration, taint *v1.Taint) bool {
+	for i := range tolerations {
+		if tolerations[i].ToleratesTaint(taint) {
+			return true
+		}
+	}
+	return false
+}
+
+type taintsFilterFunc func(*v1.Taint) bool
+
+// TolerationsTolerateTaintsWithFilter checks if given tolerations tolerates
+// all the taints that apply to the filter in given taint list.
+// DEPRECATED: Please use FindMatchingUntoleratedTaint instead.
+func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v1.Taint, applyFilter taintsFilterFunc) bool {
```

Comment:
Time to drop it given it's deprecated?

Reply:
Yeah I thought about that. Updated with it in a new commit